cli: use sensible file endings in debug zip

tbg opened the pull request:
Avoid time spent being forced by your browser to download these files
because it can't figure out which of your editors could open them.

Also add checks that prevent this naming scheme from being broken
by most future additions.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/36155)
<!-- Reviewable:end -->

danhhz reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/36155#-:-Laul43M1TIo8FbIdcXC:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @danhhz and @tbg)

___
*[pkg/cli/zip.go, line 93 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/36155#-LaukXCc3C4f-T0NPplF:-LaukXCc3C4f-T0NPplG:b4s6krd) ([raw file](https://github.com/cockroachdb/cockroach/blob/7a0de922a6ff6b48b537899eabbb020c19e72acc/pkg/cli/zip.go#L93)):*
> ```Go
> 
> func (z *zipper) createJSON(name string, m interface{}) error {
> 	if !strings.HasSuffix(name, ".json") {
> ```

can we add a similar check in `createRaw` to enforce that it has some extension (`if filepath.Ext(name) == nil { error }`) so we can't add something like a new pprof without an extension


<!-- Sent from Reviewable.io -->

tbg committed:
cli: use sensible file endings in debug zip

Avoid time spent being forced by your browser to download these files
because it can't figure out which of your editors could open them.

Also add checks that prevent this naming scheme from being broken
by most future additions.

Release note: None

tbg commented:
Done, TFTR! More PRs of this kind upcoming...

bors r=danhhz
